satellite/metainfo/loop.go

```diff
@@ -174,7 +172,6 @@ waitformore:
 				nextObservers := observers[:0]
 
 				for _, observer := range observers {
-					// TODO: move single observer handling into a separate func
```

Comment:
this todo was not copleted

Reply:
I addressed it in another comment, but since a lot of the observer logic is directly tied to this for loop, I thought it would have been clearer to just leave that logic here rather than moving it out to another function. If you still think I should move it out to another function, I will